libr/include/r_util/r_annotated_code.h

```diff
@@ -0,0 +1,123 @@
+
+#ifndef R2GHIDRA_ANNOTATEDCODE_H
+#define R2GHIDRA_ANNOTATEDCODE_H
+
+// #include <r_core.h>
+#include <r_types.h>
+#include <r_vector.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum r_syntax_highlight_type_t {
+	R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD,
+	R_SYNTAX_HIGHLIGHT_TYPE_COMMENT,
+	R_SYNTAX_HIGHLIGHT_TYPE_DATATYPE,
+	R_SYNTAX_HIGHLIGHT_TYPE_FUNCTION_NAME,
+	R_SYNTAX_HIGHLIGHT_TYPE_FUNCTION_PARAMETER,
+	R_SYNTAX_HIGHLIGHT_TYPE_LOCAL_VARIABLE,
+	R_SYNTAX_HIGHLIGHT_TYPE_CONSTANT_VARIABLE,
+	R_SYNTAX_HIGHLIGHT_TYPE_GLOBAL_VARIABLE,
+} RSyntaxHighlightType;
+
+typedef enum r_code_annotation_type_t {
+	R_CODE_ANNOTATION_TYPE_OFFSET,
```

Comment:
Maybe that's a stupid question or it's already explained elsewhere, but... What is a OFFSET annotation? What does it represent exactly? It may be worth providing a short explanation here, unless it is already obvious from somewhere else I'm missing.

Reply:
There are two types of annotation.
```
typedef enum r_code_annotation_type_t {
	R_CODE_ANNOTATION_TYPE_OFFSET,
	R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT,
	// ...
} RCodeAnnotationType;
```
By the way `RAnnotatedCode` is defined, it could represent either one of these types. See code below (some parts are not shown)
```
typedef struct r_code_annotation_t {
	RCodeAnnotationType type;
	union {
		struct {
			ut64 offset;
		} offset;
		struct {
			RSyntaxHighlightType type;
		} syntax_highlight;
	};
} RCodeAnnotation;
```
If you still think it should be explained, I will do that in `r_annotated_code.h`.

Reply:
Sorry, I misread the question. `R_CODE_ANNOTATION_TYPE_OFFSET` represent the corresponding offsets for parts of the code
For example, see an output of `pdgo`

>                   |
>                   |void main(void)
>                   |{
>     0x00001158    |    sym.imp.puts("Hello, World!");
>     0x0000115f    |    return;
>                   |}

In the `RAnnotatedCode` representing this output, there will be two annotations of type `R_CODE_ANNOTATION_TYPE_OFFSET`, they will represent give 0x00001158 and 0x0000115f.

Reply:
See the output of `pdgj` for the same code. (Some parts are not shown)

> {"start":23,"end":52,"type":"offset","offset":4440},{"start":58,"end":64,"type":"offset","offset":4447},{"start":58,"end":64,"type":"syntax_highlight","syntax_highlight":"keyword"},{"start":58,"end":64,"type":"offset","offset":4447}"

Is it clear now?

Reply:
So if I understand it right, it allows you to say which offset is associated with a given part of the decompiled code. Is that right? Maybe a small sentence explaining this would help for future devs

Reply:
I have given a short explanation as you asked me to.

Reply:
Yes, that's right.